Implementing logging (see issue #6)

asterized opened the pull request:
Implementing logging to the bot

asterized committed:
Create logger.py

Implementing logging to the bot

sexnine reviewed (changes requested):
This should not be in the same file as the logs themselves.  Code and logs in the same folder is bad idea.

Also needs docker support which I can help you with.

sexnine commented:
@SomeHybrid Are the log files split into different instances of the bot?  You don't wanna have a massive log file yk.

SomeHybrid commented:
> @SomeHybrid Are the log files split into different instances of the bot? You don't wanna have a massive log file yk.

good idea

asterized committed:
Update logger.py

Added multiple files for logging and removing the oldest file if the amount of logs is over 10.

asterized committed:
Update logger.py

Added comments to make it look better

SomeHybrid commented:
added some comments ig

asterized committed:
Delete logger.py

asterized committed:
Create current.log

asterized committed:
Create logger.py

SomeHybrid commented:
removed logger.py from logs and put it in bot.util

asterized committed:
Fixed bugs in logging. Ready for implementation.

SomeHybrid commented:
i made a stupid mistake NOOOOOOOOOOOOO

asterized committed:
fixed more bugs

sexnine commented on bot/util/logger.py:
```diff
@@ -0,0 +1,47 @@
+logging.basicConfig(filename=path + "current.log", level=logging.INFO, format='%(asctime)s - [%(levelname)s]: %(message)s')
+
+def init():
+    with open(path + "current.log", "r") as f:
```
I suggest opening the file with read and write so it doesn't have to be opened again.
```suggestion
    with open(path + "current.log", "rw") as f:
```

SomeHybrid replied:
not possible as r+ appends to the file, rw doesn't exist and w+ truncates the file before use.

sexnine commented on bot/util/logger.py:
```diff
@@ -0,0 +1,47 @@
+        data = f.read()
+        with open(path + "current.log", "w") as f1:
+            if data != "":
+                ts = data.split("\n")[0]
```
Make variable more specific, doesn't matter if it's long :)

sexnine commented on bot/util/logger.py:
```diff
@@ -0,0 +1,47 @@
+# implementing the logger class
+class Logger:
+    def __init__(self, s):
+        self.s = s
```
More specific variable please.

sexnine commented on bot/util/logger.py:
```diff
@@ -0,0 +1,47 @@
+    def __init__(self, s):
+        self.s = s
+        self.start()
+    
```
Perhaps this could be made as a simple variable so it's more clear in the rest of the code.
```suggestion
    self.cog_name = s.__class__.__name__
```

sexnine commented on logs/current.log:
```diff
@@ -0,0 +1,5 @@
+1649590457
```
Add files in logs to `.gitignore`

SomeHybrid commented:
good ideas

asterized committed:
Added implementation to logger.py (needs testing and adding CogErrorLogging to config)

asterized committed:
fixed bugs

SomeHybrid commented:
still needs testing and adding to config file

sexnine commented on bot/util/logger.py:
```diff
@@ -0,0 +1,63 @@
+        logging.info(f"Cog {cog_name} has been initialized.")
+    
+    # create the logging functions
+    def info(self, cog_name, message):
```
The cog's name shouldn't have to be passed each time it needs to log something.

asterized committed:
Update .gitignore

Added more logs to gitignore

Co-authored-by: sexnine <25860388+sexnine@users.noreply.github.com>
